blockchain/txSinger.go

```diff
@@ -0,0 +1,50 @@
+package blockchain
+
+import (
+	"github.com/ElrondNetwork/elrond-go/crypto"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/ed25519"
+	"github.com/ElrondNetwork/elrond-go/crypto/signing/ed25519/singlesig"
+)
+
+var suite = ed25519.NewEd25519()
+var singleSigner = &singlesig.Ed25519Signer{}
+
+type txSigner struct {
+	keyGen crypto.KeyGenerator
+}
+
+// NewTxSigner contains the primitives used to correctly sign a transaction
```

Comment:
this comment matches better `type txSigner struct`
rephrase to `// NewTxSigner returns a new instance of txSigner`

Reply:
done